### PR TITLE
Added `publish: unlisted` support with tests, and updated mocha.

### PR DESCRIPTION
**Pull request by WesSouza**
Added a simple use case to remove the `collection` metadata on posts, while still publishing them. This is quite useful for usage with `metalsmith-collections` and all other plugins that use it such as `metalsmith-feed`.

Also updated mocha to 2.1.0 to supress a deprecated warning from node 0.12.0.

Also added `npm test` command support.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,11 +6,14 @@
...
+  "scripts": {
+    "test": "mocha --reporter spec test/index.js"
+  },
   "dependencies": {
     "debug": "~0.7.4"
   },
   "devDependencies": {
     "metalsmith": "0.x",
-    "mocha": "1.x"
+    "mocha": "^2.1.0"
```
Comment (mikestopcontinues) [inline package.json]: This makes it too easy for dependencies to break while we're not looking. I'm OK with ~ rather than ^.

**Comment by mikestopcontinues** [comment]
Powerful move, Wes! Please just make the package.json change and I'll merge tomorrow.

**Comment by WesSouza** [comment]
Should I have bumped the version as well?

**Comment by mikestopcontinues** [comment]
Damn, I'm sorry. I had my tonsils removed Thursday, and my head has been completely MIA.

**Pull request merged**